sdk/core/Azure.Core/tests/AzureBaseBuffersExtensionsTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core.Buffers;
+using Microsoft.Identity.Client;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests
+{
+    public class AzureBaseBuffersExtensionsTests
+    {
+        [Test]
+        public async Task WriteBufferToStream()
```

Comment:
I don't think you are exercising the `!MemoryMarshal.TryGetArray` here because in all the cases memory is backed by the array.

You might need to implement a `MemoryManager` to hit it.

Reply:
Yeah, I saw that. I did verify this under the debugger though. 

As for writing a `MemoryManager` -- even in that case I am not sure if that would be enough as we are using the `MemoryMarshal` underneath and I haven't figured out a good way of replacing the manager insider that class.

Reply:
https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.IO/tests/Stream/Stream.ReadWriteSpan.cs#L116